hw/ip/otbn/dv/otbnsim/sim/edn_client.py

```diff
@@ -64,14 +64,19 @@ def forget(self) -> None:
     def take_word(self, word: int) -> None:
         '''Take a 32-bit data word that we've been waiting for'''
         assert 0 <= word < (1 << 32)
-        assert self._acc is not None
-        assert len(self._acc) < ACC_LEN
-        assert self._cdc_counter is None
+        # Ignore incoming EDN words if there is no request for that word.
+        # We might still get words if we are in the middle of an EDN request
+        # but also had a reset (in which we drop the request flag)
+        if self._acc is None:
+            return
+        else:
+          assert len(self._acc) < ACC_LEN
+          assert self._cdc_counter is None
 
-        self._acc.append(word)
+          self._acc.append(word)
 
-        if len(self._acc) == ACC_LEN:
-            self._cdc_counter = 0
+          if len(self._acc) == ACC_LEN:
+              self._cdc_counter = 0
 
```

Comment:
Nit: suggest to remove emtpy lines above `self._acc.append(word)` and `if len(self._acc) == ACC_LEN:` to keep the body of the `else` statement as one visually contiguous block.